apps/meteor/client/views/omnichannel/departments/NewDepartment.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEndpoint, useSetModal, useTranslation } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+import React from 'react';
+
+import EnterpriseDepartmentsModal from '../../../components/Omnichannel/modals/EnterpriseDepartmentsModal';
+import PageSkeleton from '../../../components/PageSkeleton';
+import EditDepartment from './EditDepartment';
+
+type NewDepartmentProps = {
+	id?: string;
+};
+
+const NewDepartment = ({ id }: NewDepartmentProps) => {
+	const setModal = useSetModal();
+	const getDepartmentCreationAvailable = useEndpoint('GET', '/v1/livechat/department/isDepartmentCreationAvailable');
+	const { data, isLoading } = useQuery(['getDepartments'], async () => getDepartmentCreationAvailable(), {
+		onSuccess: (data) => {
+			if (data.isDepartmentCreationAvailable === false) {
+				setModal(<EnterpriseDepartmentsModal closeModal={(): void => setModal(null)} />);
+			}
+		},
+	});
+
+	const t = useTranslation();
+
+	if (!data || isLoading || !data.isDepartmentCreationAvailable) {
+		return <PageSkeleton />;
+	}
```

Comment:
I believe there's a possibility of this component being stuck in a loading (skeleton) state, since we are not checking for request errors.